src/index.js

```diff
@@ -16,98 +16,54 @@ let ElasticsearchProvider = (
       not: 'must_not',
     }[group.join || 'and']
 
-    let result = {
+    return {
       bool: {
         [join]: filters,
+        ...(join === 'should' && { minimum_should_match: 1 }),
       },
     }
-    if (join === 'should') result.bool.minimum_should_match = 1
-
-    return result
   },
-  runSearch(options = {}, context, schema, filters, aggs) {
-    let query = filters
+  async runSearch({ requestorContext } = {}, node, schema, filters, aggs) {
+    let { scroll, scrollId } = node
+    let request = scrollId
+      ? // If we have scrollId then keep scrolling, no query needed
+        { scroll: scroll === true ? '2m' : scroll, scrollId }
+      : // Deterministic ordering of JSON keys for request cache optimization
+        stableKeys({
+          index: schema.elasticsearch.index,
+          // Scroll support (used for bulk export)
+          ...(scroll && { scroll: scroll === true ? '2m' : scroll }),
+          body: {
+            // Wrap in constant_score when not sorting by score to avoid wasting time on relevance scoring
+            query:
+              filters && !_.has('sort._score', aggs)
+                ? constantScore(filters)
+                : filters,
+            // If there are aggs, skip search results
+            ...(aggs.aggs && { size: 0 }),
+            // Sorting by _doc is more efficient for scrolling since it won't waste time on any sorting
+            ...(scroll && { sort: ['_doc'] }),
+            ...aggs,
+          },
+        })
 
-    // Wrapping any query NOT sorted by _score in a constant_score,
-    // so that it returns a constant score equal to the query boost
-    // for every document in the filter.
-    // Filter clauses are executed in filter context, meaning that scoring
-    // is ignored and clauses are considered for caching.
-    if (query && !_.has('sort._score', aggs))
-      query = {
-        constant_score: {
-          filter: query,
-        },
-      }
+    let requestOptions = _.merge({ headers: requestorContext }, config.request)
 
-    // Could nestify aggs here generically based on schema
-    let request = {
-      body: _.extend({ query }, aggs),
+    let client = config.getClient()
+    // If we have a scrollId, use a different client API method
+    let search = scrollId ? client.scroll : client.search
+    var response
+    try {
+      var { body: response } = await search(request, requestOptions)
+    } catch (e) {
+      response = e
+      node.error = e.meta.body
```

Comment:
Just pointing out that this is a _significant_ improvement. Errors on one node will no longer block the rest of the search, and now you'll be able to see the raw ES errors for each request in the dev tools tied to each raw ES request 🎉 

This made testing ES 7 compatibility issues substantially easier.

Reply:
We may want to consider (optionally?) logging these with `console.error` because now they won't show up in the logs anymore. The good news is that they're no longer blocking and are way more debuggable, but from an ops perspective we're kinda suppressing these errors now 🤔